src/os/im/action/importactionmanager.js

```diff
@@ -381,13 +436,132 @@ os.im.action.ImportActionManager.prototype.processItems = function(entryType, op
  */
 os.im.action.ImportActionManager.prototype.unprocessItems = function(entryType, items) {
   if (items && items.length > 0) {
-    var entries = this.actionEntries[entryType];
-    if (entries && entries.length > 0) {
-      for (var i = 0; i < entries.length; i++) {
-        entries[i].unprocessItems(items);
+    this.processItemsInternal_(entryType, items, true, true);
+  }
+};
+
+
+/**
+ * Consolidate results of desired notification(s) from multiple FeatureActions
+ *
+ * @param {os.im.action.ImportActionCallbackConfig} config
+ * @param {os.im.action.ImportActionCallbackConfig} cfg
+ * @private
+ */
+os.im.action.ImportActionManager.mergeNotify_ = function(config, cfg) {
+  if (!config) return;
+
+  if (cfg) {
+    config.labelUpdateShown = config.labelUpdateShown || cfg.labelUpdateShown;
+    config.notifyStyleChange = config.notifyStyleChange || cfg.notifyStyleChange;
+    config.setColor = config.setColor || cfg.setColor;
+    config.setFeaturesStyle = config.setFeaturesStyle || cfg.setFeaturesStyle;
+
+    if (cfg.color) {
+      // add the next colors
+      cfg.color.forEach((color) => {
+        // TODO merge same-colors into a single color entry
+        config.color.push(color); // flatten the tree
+      });
+    }
+  }
+};
+
+/**
+ * Optimize the colors to avoid overlaps (max N instead of N^2 events)
+ *
+ * @param {os.im.action.ImportActionCallbackConfig} config
+ * @private
+ */
+os.im.action.ImportActionManager.mergeNotifyColor_ = function(config) {
+  // TODO benchmark which is faster -- removing overlaps or just re-setting them when there's 25%...100% overlap
+  var len = (config) ? config.color.length : -1;
+
+  // only do this extra step when there are more than one (possibly conflicting) color actions
+  if (len > 1) {
+    var colorItemsCount = config.color.reduce((count, colorConfig) => {
+      return count + ((colorConfig[0]) ? colorConfig[0].length : 0);
+    }, 0);
+
+    // deconflicting is expensive; only do it when there are more than N items being colored
+    if (colorItemsCount > os.im.action.ImportActionManager.MIN_ITEMS_MERGE_NOTIFY_COLOR) {
+      // the item(s) whose final color is already set
+      var all = {};
+
+      // loop backwards through colors... and remove items overlap in previous entries (last one wins)
+      for (var i = (len - 1); i >= 0; i--) {
+        var ids = {};
+        var [items] = config.color[i] || [];
+
+        // map the array by ids
+        if (i > 0) { // skip when no more loops to do
+          (items || []).reduce((map, item) => {
+            map[item.id_] = true;
```

Comment:
This is also specific to `ol.Feature`. The generic item handled by this manager may not have an `id_` field. I'm curious why the compiler didn't complain about this, given it's also a private field.